rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactory.java

```diff
@@ -0,0 +1,275 @@
+import org.apache.cxf.interceptor.LoggingOutInterceptor;
+import org.apache.cxf.jaxrs.client.Client;
+import org.apache.cxf.jaxrs.client.ClientConfiguration;
+import org.apache.cxf.jaxrs.client.JAXRSClientFactoryBean;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.transport.http.HTTPConduit;
+import org.apache.shiro.subject.Subject;
+import org.apache.ws.security.WSSecurityException;
+import org.apache.ws.security.components.crypto.CredentialException;
+import org.apache.ws.security.components.crypto.Merlin;
+import org.codice.ddf.security.common.PropertiesLoader;
+import org.codice.ddf.security.handler.api.PKIAuthenticationToken;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+
+public class SecureCxfClientFactory<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecureCxfClientFactory.class);
+
+    private final Client cxfClient;
+
+    /**
+     * @see #SecureCxfClientFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public SecureCxfClientFactory(String endpointUrl, Class<T> interfaceClass)
+            throws SecurityServiceException {
+        this(endpointUrl, interfaceClass, null, null);
+    }
+
+    /**
+     * @see #SecureCxfClientFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public SecureCxfClientFactory(String endpointUrl, Class<T> interfaceClass, String username,
+            String password) throws SecurityServiceException {
+        this(endpointUrl, interfaceClass, username, password, null, false);
+    }
+
+    /**
+     * Constructs a factory that will return security-aware cxf clients. Once constructed,
+     * use the getClient* methods to retrieve a fresh client  with the same configuration.
+     * <p/>
+     * This factory can and should be cached. The clients it constructs should not be.
+     *
+     * @param endpointUrl    the remote url to connect to
+     * @param interfaceClass an interface representing the resource at the remote url
+     * @param username       optional username for basic auth
+     * @param password       optional password for basic auth
+     * @param providers      optional list of providers to further configure the client
+     * @param disableCnCheck disable ssl check for common name / host name match
+     */
+    public SecureCxfClientFactory(String endpointUrl, Class<T> interfaceClass, String username,
```

Comment:
I've been jamming it into the OpenSearch source in order to smoke-test it. It's currently a little bit fiddly in the installer, due to the security dependencies it adds to catalog. I'm working on resolving those, then I can attach a patch if you want to see it working.